tests/unit/test_analytics.py

```diff
@@ -1,6 +1,80 @@
 import pytest
+import mock
+import platform
+import json
 
-from dvc.analytics import Analytics
+from voluptuous import Schema, Any
+
+from dvc import analytics
+from dvc.cli import parse_args
+from dvc.utils.compat import str, builtin_str
+
+
+string = Any(str, builtin_str)
+
+
+@pytest.fixture
+def tmp_global_config(tmp_path):
+    """
+    Fixture to prevent modifying the actual global config
+    """
+    with mock.patch(
+        "dvc.config.Config.get_global_config_dir", return_value=str(tmp_path)
+    ):
+        yield
+
+
+@mock.patch("dvc.daemon._spawn")
+@mock.patch("json.dump")
+def test_collect_and_send_report(mock_json, mock_daemon, tmp_global_config):
+    analytics.collect_and_send_report()
+    report = mock_json.call_args[0][0]
+
+    with pytest.raises(KeyError):
+        report["cmd_class"]
+
+    with pytest.raises(KeyError):
+        report["cmd_return_code"]
+
+    args = parse_args(["add", "foo"])
+    return_code = 0
+
+    analytics.collect_and_send_report(args, return_code)
+    report = mock_json.call_args[0][0]
+
+    assert report["cmd_class"] == "CmdAdd"
+    assert report["cmd_return_code"] == return_code
+
+    assert mock_daemon.call_count == 2
+
+
+def test_runtime_info(tmp_global_config):
+    schema = Schema(
+        {
+            "dvc_version": string,
+            "is_binary": bool,
+            "scm_class": Any("Git", None),
+            "user_id": string,
+            "system_info": dict,
+        },
+        required=True,
+    )
+
+    assert schema(analytics._runtime_info())
+
+
+@mock.patch("requests.post")
+def test_send(mock_post, tmp_path):
+    url = "https://analytics.dvc.org"
+    report = {"name": "dummy report"}
+    fname = str(tmp_path / "report")
```

Comment:
Use `fspath()`.

Reply:
Same comment as this one https://github.com/iterative/dvc/pull/2826#discussion_r355509475

:disappointed: